autoload/inspecthi.vim

```diff
@@ -101,16 +101,76 @@ function! s:reserve_popup() abort
 endfunction
 
 
+function! s:show_floatwin(...) abort
+  let win_id = b:inspecthi.inspector_win_id
+
+  if win_id != 0
+    call nvim_win_close(win_id, v:true)
+  endif
+
+  let text = s:inspect()
+  if len(text) == 0
+    let text = ' '
+  endif
+
+  let buf = nvim_create_buf(v:false, v:true)
+  call nvim_buf_set_lines(buf, 0, 1, 0, [text])
+  let win_id = nvim_open_win(buf, v:false, {
+        \ 'relative': 'cursor',
+        \ 'height': 1, 'width': len(text),
+        \ 'row': 1, 'col': 1,
+        \ 'focusable': v:false,
+        \ })
+  call nvim_win_set_option(win_id, 'colorcolumn', '')
+  call nvim_win_set_option(win_id, 'list', v:false)
+  call nvim_win_set_option(win_id, 'number', v:false)
+  call nvim_win_set_option(win_id, 'relativenumber', v:false)
+  call nvim_win_set_option(win_id, 'spell', v:false)
+  call nvim_win_set_option(win_id, 'winblend', g:inspecthi_floatwin_blend)
+  call nvim_win_set_option(win_id, 'winhighlight', 'Normal:' . g:inspecthi_floatwin_hl)
+  let b:inspecthi.inspector_win_id = win_id
+endfunction
+
+
+function! s:close_floatwin() abort
+  let win_id = b:inspecthi.inspector_win_id
+  if win_id != 0
+    call nvim_win_close(win_id, v:true)
+    let b:inspecthi.inspector_win_id = 0
+  endif
+endfunction
+
+
+function! s:reserve_floatwin() abort
+  let timer_id = get(b:inspecthi, 'timer_id', 0)
+  if timer_id != 0
+    call timer_stop(timer_id)
+  endif
+
+  let b:inspecthi.timer_id = timer_start(
+        \ 500,
+        \ function('s:show_floatwin'))
+endfunction
+
+
 function! inspecthi#show_inspector() abort
   call s:init_vars()
-  call s:show_popup()
+  if has('nvim')
```

Comment:
It looks like too many `if has('nvim')` conditions. How about putting them into a function?

```vim
"...

function s:show_popup() abort
  if has('nvim')
    call s:show_popup_nvim()
  else
    call s:show_popup_vim()
  endif
endfunction

function! inspecthi#show_inspector() abort
  call s:init_vars()
  call s:show_popup()
  let b:inspecthi.shows_inspector = 1
endfunction
```